src/numbers/fixed_point/fp16/core.cairo

```diff
@@ -0,0 +1,167 @@
+use core::debug::PrintTrait;
+
+use orion::numbers::fixed_point::core::FixedTrait;
+use orion::numbers::fixed_point::implementations::fp16x16::math::{
+    core as core_math, trig, hyp, erf
```

Comment:
these modules have functions implemented for FP16x16. Not the fp16 implementation. All the functions in that modules needs to be re-implemented for fp16 as well, otherwise you cannot import them in the `fp16Impl`